text/0000-app-image-extensions.md

```diff
@@ -21,48 +22,139 @@ Allowing buildpacks to install OS packages dynamically during build would drasti
 Mixins already allow buildpack authors to create buildpacks that depend on an extended set of OS packages without affecting build time.
 However, it is not uncommon for application code to depend on OS packages.
 
-Given that app authors and platform maintainers experience increased build time directly, extending the stack at the app author's or platform maintainer's request may add flexibility without sacrificing performance.
+Given that app authors and platform maintainers experience increased build time directly, extending the stack at the app author's request may add flexibility without sacrificing performance. At the same time, we want the CNB interface to be highly orthogonal, which is why we've strived to "make everything a buildpack". Allowing some buildpacks to run with privileges would give users the flexibility they expect based on their experience with other tools like `Dockerfile`.
+
+The advantages of using a buildpack for privileged operations are the same as for unprivileged operations: they are composable, fast (caching), modular, reuseable, and safe.
+
+As an example, consider the use case of Acme.com. They have three teams that want to contibute privileged layers to every image built at the company. Each of these teams wants to manage their own artifacts and the mechanisms for installing them, but none of these teams are in control of the base image. The teams have special logic for when their layers should be added to an image; in some cases it's the precense of a Node.js app, and in others it's the use of Tomcat, etc. In the past, these teams were trying to contribute their layers by selectively adding them to every `Dockerfile` for every base image in the company, but this doesn't scale well. With Root Buildpacks, each team can manage their own artifacts, release cadence, and add logic to `bin/detect` to ensure the layers are added when needed.
 
 # What it is
 [what-it-is]: #what-it-is
 
-## Specification
+- *root buildpack* - a special case of buildpack that is run with privileges
+
+Application developers can use root buildpacks to extend their build and/or run images.
+
+We introduce a boolean `privileged` key in the `[buildpack]` table of `buildpack.toml`, which is defined as follows:
+
+```
+[buildpack]
+privileged = <boolean (default=false)>
+```
+
+When `privileged` is set to `true`, the lifecycle will run this buildpack as the `root` user.
 
-Stack image creators may add the following executables to their run and/or build images:
+For each root buildpack, the lifecycle will use [snapshotting](https://github.com/GoogleContainerTools/kaniko/blob/master/docs/designdoc.md#snapshotting-snapshotting) to capture changes made during the buildpack's build phase (excluding `/tmp`, `/cnb`, and `/layers`). All of the captured changes will be included in a single layer produced as output from the buildpack. The `/layers` dir MAY NOT be used to create arbitrary layers.
```

Comment:
Will pack and platforms be expected to add the lifecycle to the run image to execute the run image buildpacks?

Reply:
Also, will snapshotting be necessary on the build image? Could the "root buildpacks" run before the regular buildpacks in the same container without needing to create an ephermeral build image?